spiffe/tls_verify.go

```diff
@@ -28,6 +28,16 @@ func VerifyPeerCertificate(peerChain []*x509.Certificate, trustDomainRoots map[s
 		return nil, err
 	}
 
+	if peer.IsCA {
+		return nil, errors.New("cannot validate peer which is a CA")
+	}
+	if peer.KeyUsage != x509.KeyUsageCertSign {
```

Comment:
`KeyUsage` is a bit field. As written, this will return an error if the `KeyUsage` bit field is not empty except the `KeyUsageCertSign` bit...

Shouldn't this be something like:
```
if peer.KeyUsage&x509.KeyUsageCertSign > 0 {
    return nil, errors.New("cannot validate peer with keyCertSign")
}
```